### ISSUE TITLE
Add summary view (?)

### ISSUE THREAD

**Issue by arrowtype** [source-foundry/fdiff#18]
I just tried fdiff for the first time. Super easy to install and use! Really nice work.

I'm not sure whether this makes sense for you to add, but I thought I'd share something that I wish I could do right now: starting at a high level to understand where diffs might be in two fonts, and then "zooming in" to inspect those more closely. Basically, I wish fdiff have more of a summary view of some kind. 

My use case is this: I changed the axis label in Recursive from "Proportion" to "Monospace," but tried to change nothing else. However, the one with the same axis called "Monospace" is about 1% bigger in filesize. This isn't very significant, but I'd still like to understand why the filesize changed. I guess that somehow, the label adjustment reoriented my GSUB features in such a way that affected the fvar deltas? However, there are too many changed lines for the fdiff view for me to make sense of.

["Before" font with PROP axis](https://github.com/arrowtype/recursive/blob/5330cc81d1a8b954712388b5505454169df07347/font-betas/recursive-prop_xprn_weight_slnt_ital--2019_09_26.ttf)

["After" font with MONO axis](https://github.com/arrowtype/recursive/blob/5330cc81d1a8b954712388b5505454169df07347/font-betas/rec-mono_xprn_wght_slnt_ital--2019_09_26.ttf)

Here's a very rough, inaccurate sketch of what I'm imagining this summary view might be:

```
▶️ fdiff --summary font-betas/recursive-prop_xprn_weight_slnt_ital--2019_09_26.ttf font-betas/rec-mono_xprn_wght_slnt_ital--2019_09_26.ttf

BEFORE: recursive-prop_xprn_weight_slnt_ital--2019_09_26.ttf (595 KB)

AFTER:  rec-mono_xprn_wght_slnt_ital--2019_09_26.ttf (598 KB, +1.005%)

Diffs in tables:

  GSUB: -189 lines
        +213 lines

  gvar: -348 lines
        +549 lines

  name: -3 lines
        +3 lines
```

Basically, I think I just need enough information for me to make sense of what tables have changed, so that I can then use the `--include` option to look at one table at a time. Honestly, I'm not quite sure that this would get me all the way to understanding, or whether there needs to be some other view, like a "medium-level" view, or something I haven't yet imagined.

Of course, it's also possible that I'm trying to use this tool for something outside its intent and that you shouldn't try to satisfy this use case. Just thought I'd report my experience, in case it's helpful.

**Comment by chrissimpkins** [comment]
Thanks for this request Stephen!  I really like this idea and am thinking through how to implement something along these lines.  As it currently stands, the diff output does not *necessarily* include the XML tag for the table header to indicate what table(s) include diffs.  We could consider adding support for automated dumps of each individual table as a separate XML file and then diff across those.  I believe that would get to the summary information that you are looking for here.   Let me give this some more thought and I'll see what I can come up with.

As a workaround until then, I think that you should be able to dump separate diffs for each table using individual calls with `--include` and run the diff output through awk to parse the number of additions and subtraction lines:

```
$awk '{count[$1]++} END {print "+++ " count["+"]-1"\n--- " count["-"]-1}' [DIFF OUTPUT PATH]
```

Adapted from information in this SO thread: https://stackoverflow.com/questions/27986425/using-awk-to-count-the-number-of-occurrences-of-a-word-in-a-column

**Comment by chrissimpkins** [comment]
Edited above information on awk as I don't think that I had the original parsing approach correct.  I don't believe that you need to exclude the initial two header lines because they should not be parsed as counts for `+` and `-` (I think?)... Requires some testing.

**Comment by chrissimpkins** [comment]
For the file size summary information in this part of your request:

```
BEFORE: recursive-prop_xprn_weight_slnt_ital--2019_09_26.ttf (595 KB)

AFTER:  rec-mono_xprn_wght_slnt_ital--2019_09_26.ttf (598 KB, +1.005%)
```

That is not a problem at all to implement.  I think the question is whether it belongs in a diff tool.  Thinking on this one.

**Comment by justvanrossum** [comment]
I wrote something today that overlaps somewhat with the ideas in this issue:

https://github.com/justvanrossum/fbdiff ("font binary diff")

It's not really a diff tool: I just needed a quick report which tables are different at all, at the binary level. More detailed diffing can then follow with `fdiff` on a per-table basis.

It may be a nice complement to `fdiff`.

**Comment by chrissimpkins** [comment]
> I wrote something today that overlaps somewhat with the ideas in this issue:
> 
> https://github.com/justvanrossum/fbdiff ("font binary diff")
> 
> It's not really a diff tool: I just needed a quick report which tables are different at all, at the binary level. More detailed diffing can then follow with `fdiff` on a per-table basis.
> 
> It may be a nice complement to `fdiff`.

This looks great and will be very helpful Just!  Thanks for sharing it here.  It looks like this *may* address Stephen's request here and this is definitely functionality that I would use. I will plan to add a link to it on the README.  This will be a good high level -> granular level diff workflow. 👍

**Comment by chrissimpkins** [comment]
@arrowtype Stephen, mind having a look at Just's tool in https://github.com/source-foundry/fdiff/issues/18#issuecomment-622078424 and let me know if that addresses your request?  If so, we will close this thread when I add the link to the README.

**Comment by justvanrossum** [comment]
Thanks! FWIW, I uploaded `fbdiff` to PyPI this morning.

**Comment by chrissimpkins** [comment]
@arrowtype Closing this as I think that Just's tool addresses your request.  Feel free to reopen if there is anything that we can do here.

**Comment by arrowtype** [comment]
Amazing; thanks for making a summary tool, @justvanrossum! I saw it on Twitter the other day, and have been meaning to try it. This is good to close, and a good reminder to try fbdiff soon.
